gax/src/main/java/com/google/api/gax/tracing/NoopApiTracerFactory.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google LLC nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.google.api.gax.tracing;
+
+import com.google.api.core.InternalApi;
+
+/**
+ * Factory that will build fake {@link ApiTracer}s.
+ *
+ * <p>For internal use only.
+ */
+@InternalApi
+public final class NoopApiTracerFactory implements ApiTracerFactory {
+  /** {@inheritDoc} */
+  @Override
+  public ApiTracer newTracer(SpanName spanName) {
+    return NoopApiTracer.create();
+  }
+
+  @Override
+  public int hashCode() {
+    return 1;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    return obj instanceof NoopApiTracerFactory;
```

Comment:
Both `equals()` and `hashCode()` look odd. I feel the proper way to achieve "same result" (all `NoopApiTracerFactory` are equal) is to have private constructor here and getInstance static method (i.e. converting this to singleton). 

Also usage of `instanceof` makes it look even "odder", since rigorous implementations of equals is supposed to compare actual classes (`getClass() == obj.getClass()`) instead of `instanceof` (yes, `instanceof` is used sometimes, for example in Hibernate entities `equals()` implementations, because in real life scenarios we would have to compare instance with proxy instance, but it is a special case). It is a final class, so `instanceof` is effectively same as comparing the actual class... Well what I mean by all this mutter is that the two methods look odd from all angles, and I would be nice to not have them at all.

Reply:
done